### ISSUE TITLE
failed scheduled step stop

### ISSUE THREAD

**Issue by aliciamchen** [empiricaly/empirica#475]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Been getting this error along with #473 and  #468. Have also been getting missing batches and players in the admin console, and participants reporting that they are not able to continue the game. 

I was wonder, perhaps I need to scale up my droplet? But I'm running a `8 GB Memory / 4 Intel vCPUs / 25 GB Disk` DigitalOcean droplet, and my CPU usage is <40%, load < 2.0, memory < 20%.

### Empirica Version

```text
Version: v1.8.12
SHA:     656d19b
Build:   159
Branch:  main
Time:    2023-12-26T07:14:40Z
```

### Relevant log output

```shell
18:50:17.204 ERR runtime: failed scheduled step stop error="invalid transition: from state mismatch with previous state" stepID=01HJP6NHKW0VQWE2SK9R7ZD7AR
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by npaton** [comment]
I think these aae different problems. We have had some issues on the client where `player.stage` was undefined and that's often resulted in the page no loading for the player, that could be the issue with participants not being able to continue the game. That should be fixed in the latest release (1.9.0).

The "invalid transition" log is not a problem and I removed the log in the latest release.

As for the `missing batches and players in the admin console`, I don't quite understand that one. Could you help me understand what you mean by that? Thanks!

**Comment by aliciamchen** [comment]
Thank you! With missing batches / players, I'm referring to #469, where the players in the admin console disappear.
